numpy/core/tests/test_dtype.py

```diff
@@ -1364,13 +1364,25 @@ def test_metadata(self):
 
     @pytest.mark.parametrize("DType",
         [type(np.dtype(t)) for t in np.typecodes['All']] +
-        [np.dtype(rational), np.dtype])
-    def test_pickle_types(self, DType):
+        [type(np.dtype(rational))])
+    def test_pickle_metaclasses(self, DType):
```

Comment:
I'm still a big confused on the terminology here, so let me know if the test names make sense.

Reply:
It is confusing, but the metaclass would be `type(type(np.dtype("i8"))` or in other words `type(np.dtype)`.  The dtype like `np.dtype("i8")` is an instance. The DType type/class is a subclass of `np.dtype`.
You could `test_pickle_dtype` and `test_pickle_dtype_class` or `dtype_type` if you prefer.

(Basicaly, there is for example an `Int64DType` class of which `"<i8"` and `">i8"` are instances.  But for user DTypes that class is created by NumPy, so it is easy to not realize it's existance.

Reply:
Thanks; I renamed the test cases and replaced `DType` with `dt` when referring to the dtype instance.